packages/react-components/src/components/status-timeline/alarmTransforms.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  COMPARISON_OPERATOR,
+  type DataPoint,
+  type DataStream,
+} from '@iot-app-kit/core';
+import type { AlarmData } from '../../hooks/useAlarms';
+import {
+  parseAlarmStateAssetProperty,
+  PascalCaseStateName,
+} from '../../hooks/useAlarms/transformers';
+
+export const transformAlarmStateToDataStream = (
+  alarm: AlarmData
+): DataStream<PascalCaseStateName> => {
+  const data = (
+    alarm.state?.data?.map((d) => {
+      const parsedData = parseAlarmStateAssetProperty(d);
+      if (parsedData) {
+        return {
+          x: parsedData?.timestamp,
+          y: parsedData?.value.state,
+          quality: parsedData?.quality,
+        };
+      }
+    }) ?? []
+  ).filter((point) => !!point) as DataPoint<PascalCaseStateName>[];
+
+  const firstInputProperty = alarm.inputProperty?.at(0)?.property;
+
+  return {
+    id: `${alarm.assetId}---${alarm.compositeModelId}---state`,
+    data: data,
+    name: alarm.compositeModelName,
+    isLoading: alarm.status.isLoading,
```

Comment:
How about `isError`? And surfacing the custom error that alarms failed to load? Not sure how this is handled internally in StatusTimelineBase.

Reply:
nice catch! updated :)